### PR TITLE
gh-52551: Use wcsftime() to implement time.strftime() on Windows

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-52551

### CODE REVIEW

File: Modules/timemodule.c [hunk 1]
```diff
@@ -740,8 +740,7 @@ checktm(struct tm* buf)
 }
 
 #ifdef MS_WINDOWS
-   /* wcsftime() doesn't format correctly time zones, see issue #10653 */
-#  undef HAVE_WCSFTIME
+#  define HAVE_WCSFTIME
```
Comment (vstinner) [inline Modules/timemodule.c]: There are compiler warnings: 'HAVE_WCSFTIME': macro redefinition [D:\a\cpython\cpython\PCbuild\_freeze_module.vcxproj]

**Comment by vstinner** [comment]
Is https://github.com/python/cpython/issues/54862 issue solved on Windows?

File: Modules/timemodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/timemodule.c]: Oh, I did not find where it was defined in C.

**Comment by serhiy-storchaka** [comment]
> Is #54862 issue solved on Windows?

What exactly is that issue about? I get no mojibake:

```pycon
>>> import time
>>> time.strftime('%Z')
'Фінляндія (літо)'
>>> time.tzname
('Фінляндія (зима)', 'Фінляндія (літо)')
```
But it crashes on invalid format string (like `%f`).

**Comment by serhiy-storchaka** [comment]
> But it crashes on invalid format string (like `%f`).

Well, this was unrelated.

**Comment by vstinner** [comment]
> Is https://github.com/python/cpython/issues/54862 issue solved on Windows?

I changed my Windows language to French and the timezone to UTC+14, I managed to get a non-ASCII timezone name:

```
vstinner@WIN C:\victor\python\main>python
Python 3.14.0a1+ (heads/pr/125658:0e1bdf72d94, Oct 18 2024, 16:46:59) [MSC v.1941 64 bit (AMD64)] on win32
>>> import time; time.strftime("%Z")
'Îles de la Ligne '
```

So at least for this locale (French_France.1252), **it works as expected.**

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by serhiy-storchaka** [comment]
Thank you for your review @vstinner.

**Pull request merged**